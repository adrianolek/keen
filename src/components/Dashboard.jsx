import React, {Component} from 'react';
import {Dataviz} from 'keen-js';

class Dashboard extends Component {

  componentDidMount() {
    this.props.keenQuery();
    this.chart = new Dataviz();
    this.chart
      .el('#chart-dataviz')
      .attributes({
        chartType: 'area',
        stacked: true
      })
      .prepare();
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.data) {
      this.chart.data(nextProps.data).render();
    }
  }

  render() {
    return (
      <div className="keen-dashboard">
        <div className="container-fluid">
          <div className="row">
            <div className="col-sm-6">
              <div className="chart-wrapper">
                <div className="chart-title">
                  Pageviews by browser (past 24 hours)
                </div>
                <div className="chart-stage" id="chart-dataviz">
                </div>
                <div className="chart-notes">
                  This is a chart rendered with Keen Dataviz.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

export default Dashboard;
