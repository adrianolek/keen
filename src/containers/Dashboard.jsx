import Dashboard from '../components/Dashboard';
import {connect} from 'react-redux';
import {keenQuery} from "../actions";

const mapStateToProps = ({data}) => ({
  data
});

const mapDispatchToProps = (dispatch) => ({
  keenQuery: () => dispatch(keenQuery())
});

export default connect(mapStateToProps, mapDispatchToProps)(Dashboard);
