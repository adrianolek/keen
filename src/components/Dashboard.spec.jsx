import React from 'react';
import {mount} from 'enzyme';
import Dashboard from "./Dashboard";

jest.mock('keen-js', () => {
  return {
    Dataviz: jest.fn(() => ({
      el: jest.fn().mockReturnThis(),
      attributes: jest.fn().mockReturnThis(),
      prepare: jest.fn().mockReturnThis(),
      data: jest.fn().mockReturnThis(),
      render: jest.fn().mockReturnThis()
    }))
  };
});

describe('Dashboard', () => {
  afterAll(() => {
    jest.unmock('keen-js');
  });

  it('queries Keen, prepares chart and renders data into DOM element', () => {
    const query = jest.fn();

    const wrapper = mount(<Dashboard keenQuery={query}/>);

    expect(query.mock.calls).toHaveLength(1);

    expect(wrapper.instance().chart.el.mock.calls[0][0]).toMatchSnapshot();
    expect(wrapper.instance().chart.prepare.mock.calls).toHaveLength(1);

    wrapper.setProps({bar: 'baz'});

    expect(wrapper.instance().chart.render.mock.calls).toHaveLength(0);

    wrapper.setProps({data: 'foo'});

    expect(wrapper.instance().chart.data.mock.calls[0][0]).toBe('foo');
    expect(wrapper.instance().chart.render.mock.calls).toHaveLength(1);
  });
});
