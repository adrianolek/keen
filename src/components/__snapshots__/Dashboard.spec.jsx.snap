// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard queries Keen, prepares chart and renders data into DOM element 1`] = `
<div
  class="chart-stage"
/>
`;
